src/meta/Base.ts

```diff
@@ -26,11 +24,18 @@ export class Base {
 		this._invalidating = global.requestAnimationFrame(this._invalidate);
 	}
 
-	protected requireNode(key: string): void {
-		this._requiredNodes.add(key);
-
-		if (!this.nodes.has(key)) {
-			this.invalidate();
+	protected requireNode(key: string, callback?: WidgetMetaRequiredNodeCallback): void {
+		const node = this.nodes.get(key);
+		if (node) {
+			callback && callback.call(this, node);
+		}
+		else {
+			const callbacks = this._requiredNodes.get(key) || [];
+			callback && callbacks.push(callback);
+			this._requiredNodes.set(key, callbacks);
+			if (!callback) {
+				this.invalidate();
```

Comment:
i guess i'm confused as to when we don't want it to be invalidated. if we call requireNode, regardless of what the callback does, things aren't going to work without that node, right? Would you mind posting a use case of this with the callback that doesn't necessarily call invalidate? That might help me understand! Thanks!